### DISCUSSION TITLE
push the missing data to my collector?

### DISCUSSION THREAD

**Discussion by mabed-fr** [netdata/netdata#14241] [Q&A]
Hello,

I use NETDATA with go modules to retrieve all the metrics of my machines (60 VMs) I find that netdata stores about 2 days of metrics locally.

I have suffered several incidents on the side of my collector (victoria metrics) Can I push the missing data to my collector?

Thank you very much

**Comment by andrewm4894** [comment]
hello @mabed-fr 

Are you looking to push data from some netdata agents to somewhere else or into some netdata agents from someother source where you have historic data?

**Reply by mabed-fr** [reply]
Thank you for your answer, I see that it is not clear, so I will argue

I have 60 machines where netdata is installed and I have victoriametrics as a collector (prometheus ++++)
It happens that my victoria metrics is down and therefore during this time it can no longer recover the data exposed by netdata, but knowing that netdata keeps this data is it possible to push this data?

DOWN Exemple and result on my graph

![image](https://user-images.githubusercontent.com/25350431/211833415-4be31d09-720d-45fc-9f08-462516bc4e25.png)

**Reply by andrewm4894** [reply]
ah i see. i think maybe you might need to see if you can somehow "replay" collection by victoriametrics perhaps. 

if you go to the netdata agent dashboard or cloud dashboards are you able to see data during that gap (to be sure its actually in the netdata agent db's)? 

if so then maybe its things in here you might need to try perhaps? https://docs.victoriametrics.com/#backfilling

**Reply by mabed-fr** [reply]
Yes the data is present in netdata local dashboard.

I'm reading link

**Reply by mabed-fr** [reply]
There are several possible formats for backfilling:

GZIP
json
single line
multi line

It's possible to export range of data from netdata local db ?

**Reply by andrewm4894** [reply]
One option could be to write some sort of script that loops over your netdata agent data and just sends to VictoriaMetrics as csv via https://docs.victoriametrics.com/#how-to-import-csv-data

So would need to loop over each chart and for some windows of time just pull data from local netdata agent api, wrangle it how you need and then just post to VictoriaMetrics via their import api. 

eg loop over charts and windows and grab data like this: https://london.my-netdata.io/api/v1/data?chart=system.cpu&format=csv&after=-900&before=0

I'm not quite seeing any other obvious ready made solution - @MrZammler or @ilyam8 do you know if there could be some way to leverage one of the [Netdata exporting engines](https://learn.netdata.cloud/docs/agent/exporting) to maybe help here?

**Reply by mabed-fr** [reply]
Can i request a specific date and hour with API ? and loop on ?

**Reply by andrewm4894** [reply]
yep `after` and `before` can be timestamps eg here is 2 hours from some time today: https://london.my-netdata.io/api/v1/data?chart=system.cpu&format=csv&after=1673442518&before=1673449738

**Reply by mabed-fr** [reply]
What format is the timestamp ?

**Reply by andrewm4894** [reply]
unix timestamps - stick them in here: https://www.epochconverter.com/

**Reply by mabed-fr** [reply]
Do I have to loop on each chart? or is there something other than /data?

**Reply by andrewm4894** [reply]
yes you would have to loop on each chart and /data can only do one chart at a time. 

i have been meaning to push for a dataframe api that would improve this (my [feature request](https://github.com/netdata/netdata/issues/14040) is here). 

If you are using Python we do have a sort of wrapper/help lib that could take some of the pain out of it: https://github.com/netdata/netdata-pandas

It will let you make requests to get all charts and it will async do the requests for you and give you back a pandas df and so can be a bit easier to work with if using Python.

**Reply by mabed-fr** [reply]
Thank you for your help ! I write the script in **bash** for first and i look for best and optimize language

**Reply by mabed-fr** [reply]
I need a prometheus format but is not available :/

**Reply by andrewm4894** [reply]
i dont think there really is a way to display a window of time in Prometheus format since Prometheus is usually just a single timestep and metrics get scraped from it. 

eg this is prom format - see only one timestamp: https://london.my-netdata.io/api/v1/allmetrics?format=prometheus

you could easily use bash to try and loop over each row of data and emit the relevant prom format easy enough i'd imagine.

**Reply by mabed-fr** [reply]
Is very different format :/
